src/affinder/load_tf.py

```diff
@@ -0,0 +1,18 @@
+import numpy as np
+from magicgui import magic_factory
+from napari.layers import Layer
+
+
+@magic_factory(call_button='Load')
+def load_affine(layer: Layer, affine: str):
```

Comment:
Ah, this is the issue: don't annotate affine as str, but as a pathlib.Path. Then magicgui will give you a nice file dialog. See the [main widget](https://github.com/jni/affinder/blob/40370a1ba81fa22db598906fa5e3fd4f40a60f3f/src/affinder/affinder.py#L138) for an example, though of course in this case you wouldn't make it optional.

In this case you also need an existing file, but for your own reference, see how to modify the behaviour of the widgets [a few lines above that](https://github.com/jni/affinder/blob/40370a1ba81fa22db598906fa5e3fd4f40a60f3f/src/affinder/affinder.py#L127).

Reply:
I actually thought about this but didn't try it in the end. Seemed to simple to be true  ^^
Thanks for the tip!

Reply:
It's called magicgui after all! 😃